copylot/hardware/lights/voltran/voltran.py

```diff
@@ -0,0 +1,516 @@
+
+    def turn_on(self):
+        """
+        Alias function that toggles Laser Emission ON
+        """
+        logger.info('Turning laser: ON')
+        self.toggle_emission = 1
+        return self._toggle_emission
+
+    def turn_off(self):
+        """
+        Alias function that toggles Laser Emission Off
+        """
+        logger.info('Turning laser: OFF')
+        self.toggle_emission = 0
+        return self._toggle_emission
+
+    @property
+    def laser_power(self):
+        """
+        Sets the laser power 
+        Requires pulse_mode() to be OFF
+        """
+        return self._curr_power
+
+    @laser_power.getter
+    def laser_power(self):
+        """
+        Sets the laser power 
+        Requires pulse_mode() to be OFF
+        """
+        self._curr_power = float(self.write_cmd('?LP')[0])
+        return self._curr_power
+
+    @laser_power.setter
+    def laser_power(self, power: float):
+        """
+        Sets the laser power 
+        Requires pulse_mode() to be OFF
+        """
+        if self._max_power is None:
+            self.maximum_power
+        if power > self._max_power:
+            power = self._max_power
+            logger.info(f'Maximum power is: {self._max_power}')
+        logger.info(f'Setting power: {power}')
+        self._curr_power = float(self.write_cmd('LP', power)[0])
+
+    @property
+    def pulse_power(self):
+        """
+        Pulse Power configuration
+        Requires pulse_mode() to be ON
+        """
+        return self._pulse_power
+
+    @pulse_power.getter
+    def pulse_power(self):
+        """
+        Pulse Power configuration
+        """
+        self._pulse_power = float(self.write_cmd('?PP')[0])
+
+    @pulse_power.setter
+    def set_pulse_power(self, power):
+        """
+        Pulse Power configuration
+        """
+        logger.info(f'Setting Power:{power}')
+        self._pulse_power = float(self.write_cmd('PP', str(power))[0])
+
+    @property
+    def pulse_mode(self):
+        """
+        Toggle Pulse Mode On and Off (1=On)
+        """
+        return self._pulse_mode
+
+    @pulse_mode.getter
+    def pulse_mode(self):
+        """
+        Toggle Pulse Mode On and Off (1=On)
+        """
+        self._pulse_mode = self.write_cmd('?PUL')[0]
+
+    @pulse_mode.setter
+    def set_pulse_mode(self, mode=0):
+        """
+        Toggle Pulse Mode On and Off (1=On)
+        """
+        self._pulse_mode = self.write_cmd('PUL', str(mode))[0]
+
+    @property
+    def maximum_power(self) -> float:
+        """
+        Request Maximum Laser Power 
+        """
+        return self._max_power
+
+    @maximum_power.getter
+    def maximum_power(self):
+        """
+        Request Maximum Laser Power 
+        """
+        self._max_power = float(self.write_cmd('?MAXP')[0])
+
+    def _echo_off(self):
+        self.write_cmd('ECHO', 0)
+        logger.debug('Echo Off')
+
+    def __enter__(self):
```

Comment:
do we need a `__enter__` implementation?

Reply:
There is no specific need for this and it may be redundat. I just added it in case someone wanted to use it within a context. The `__exit__` will deal with the cleanup.

Reply:
I think we should promote a single API and for now we can remove it. If there is demand for using this class within a context manager, we can have these back again.

Reply:
I removed it and it's ready on [04cffcc](https://github.com/czbiohub/coPylot/pull/148/commits/04cffcc53f2925aff1613e78e1b87f70d1fb7497)